HIP/hip-xxx.md

```diff
@@ -0,0 +1,140 @@
+## hip: <HIP number (assigned by the HIP editor), usually the PR number>
+title: Enhance mirror node rest API to query by token name
+author: Simi Hunjan <SimiHunjan>, Ali Nikan <@alinik4n>
+working-group: Steven Sheehy <@steven-sheehy> Eric <ericleponner> Simon <svienot>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Draft
+created: 2024-04-19
+updated: 2024-05-14
+superseded-by: Hashport, Bonzo Finance
+
+## Abstract
+
+This HIP provides the capability of querying tokens by token name in mirror node REST API and getting the corresponding token ID information.
+
+## Motivation
+
+To enhance user experience and allow blockchain explorers such as Hashscan to enable the capability of searching by token name, we want to add this functionality to mirrornode to support querying tokens by token name.
+This is a very common use case across other explorers in the larger ecosystem, and will allow retail users and developers to be able to search for a token by token name instead of token ID.
+
+## Rationale
+
+Many times a user may not know the exact token ID when searching for a token, but they know the name. This capability allow users to search by the name and will return results when there is a partial match or exact match. Community members from bridges, Dexes, etc. have been requesting for this for a while.
+
+## User stories
+
+- As a user of mirror node, I want to be able to query with a token name (e.g Sauce) and get the corresponding token ID and information.
+- As a user of mirror node, I want to be able to see full match and partial match to my search. (e.g when querying for Sauce, I want to see result for $sauce and $xsauce)
+- As a user of a Hedera explorer, I want to look up a token by searching for token name.
+
+## Specification
+
+The proposed enhancement involves adding  query parameters `name` and `sortby` to the existing route `/api/v1/tokens` to return information about a token.
+
+**Multiple Query Results**
+
+Names assigned to tokens when tokens are created on the network are not unique. Therefore, many tokens with the same name can exist. Querying a token by a given name will return all the tokens that have the specified name defined in the token name.
+
+**Name Query Parameter**
+
+It enables to specify value that token names should match. The following form can be specified:
+
+- name={value}
+- name={op}:{value} where op is one of the usual operators (`eq`,`ne`,`lt`,`lte`,`gt`,`gte`)
```

Comment:
Remove all operators. We cannot support any of those except equals and since that's the default we don't need any `op`. Implicitly `name=foo` will search for it anywhere in the string (starts with, ends with, contains) case insensitively.

Reply:
Updated, let me know if it's accurate now